File: Seojin/시소 짝꿍/시소 짝꿍.java
import java.util.*;
class Solution {
    public long solution(int[] weights) {
        long answer = 0;
        Arrays.sort(weights);
        
        //4 2, 3 2, 4 3
        for (int i=0;i<weights.length-1;i++){
            int cur = weights[i];
            for (int j=i+1;j<weights.length;j++){
                if (cur == weights[j]) answer++;
                else if (cur*4 == weights[j]*2) answer++;
                else if (cur*3 == weights[j]*2) answer++;
                else if (cur*4 == weights[j]*3) answer++;
            }
        }
        return answer;
    }
}

File: Suho/시소 짝꿍/시소 짝꿍.py
def solution(weights):
    answer = 0
    # 270/360 = 3/4
    # 1 , 2/3, 3/4, 1/2 중 하나면 됨
    v = [1,2/3,3/4,1/2] 
    weights.sort()
    #print(weights)
    count = {} # 지나온 몸무개 갯수
    
    for i in range(len(weights)):
        weights_v = []
        for j in range(4):
            weights_v.append(weights[i]*v[j])
        #print(weights_v)
        for candidate in weights_v:
            #count배열에 후보군이 없으면 0
            answer += count.get(candidate, 0)
        count[weights[i]] = count.get(weights[i], 0) + 1
        #print("count=",count)
        
    return answer